controller_manager/src/controller_manager.cpp

```diff
@@ -120,8 +858,8 @@ controller_interface::return_type
 ControllerManager::activate() const
 {
   auto ret = controller_interface::return_type::SUCCESS;
-  for (auto loaded_controller : loaded_controllers_) {
-    auto controller_state = loaded_controller->get_lifecycle_node()->activate();
+  for (auto loaded_controller : controllers_lists_[current_controllers_list_]) {
+    auto controller_state = loaded_controller.c->get_lifecycle_node()->activate();
```

Comment:
I believe this code should be completely removed now.

When we activate the controller manager, we should not activate all loaded controllers, we should only activate controllers on the `switch_controller` function.

But for deactivate, I believe we need to deactivate them.

This creates a unintuitive situation, you deactivate the controller manager, it deactivates the controllers, but you activate it, and it doesn't activate the controllers.

Reply:
think of deactivating the controller manager as a last resort, something that an emergency stop would call for instance. I agree this part should be gone

Reply:
What about activating previously active controllers? 

Like you said, in the event of an emergency (which happen quite often with research robots) it's going to be a pain in the ass to reactivate controllers manually, it's something that is not needed in ROS1 right now.

Another alternative is not to deactivate the controllers themselves, just stop updating them because the controller manager is disabled.

Reply:
I think this one is strongly about safety. A velocity controller will keep sending the same velocity (arm of robot base!) without an update, robot arms may be restarted with the old command but from a new joint position (if an arm doesn't know to hold position without sending command actively) and may overreact to reach the new target. I'd err on the side of caution and don't mind people writing one-liners to call the switch functionality every time they had to shut down controls.

Reply:
All right, removing the activation of controllers.